### PR TITLE
farsight-security-57 Add support for DNSDB API version 2

### PR DESCRIPTION
**Pull request by hstern**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Adds support for DNSDB API version 2 with Flexible Search

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/DNSDB/Integrations/DNSDB_v2/README.md [hunk 1]
```diff
@@ -16,22 +16,123 @@ lookup rich, historical DNS information – either as playbook tasks or through
 API calls in the War Room – to access rdata and rrset records.  It was
 integrated and tested with version 1 of the DNSDB API.
 
-## Configure DNSDB on Cortex XSOAR
+## Configure DNSDB_v2 on Cortex XSOAR
 
 1. Navigate to **Settings** > **Integrations** > **Servers & Services**.
-2. Search for DNSDB.
+2. Search for DNSDB_v2.
```
Comment (ShahafBenYakir) [inline Packs/DNSDB/Integrations/DNSDB_v2/README.md:22]:
```suggestion
2. Search for DNSDB v2.
```

File: Packs/DNSDB/Integrations/DNSDB_v2/README.md [hunk 2]
```diff
@@ -16,22 +16,123 @@ lookup rich, historical DNS information – either as playbook tasks or through
 API calls in the War Room – to access rdata and rrset records.  It was
 integrated and tested with version 1 of the DNSDB API.
 
-## Configure DNSDB on Cortex XSOAR
+## Configure DNSDB_v2 on Cortex XSOAR
```
Comment (ShahafBenYakir) [inline Packs/DNSDB/Integrations/DNSDB_v2/README.md:19]:
```suggestion
## Configure DNSDB v2 on Cortex XSOAR
```

File: Packs/DNSDB/Integrations/DNSDB_v2/DNSDB_v2.py [hunk 3]
```diff
@@ -438,7 +537,7 @@ def lookup_to_markdown(results: List[Dict], want_bailiwick=True) -> str:
         row['FromZoneFile'] = str("zone_time_first" in result)
         out.append(row)
 
-    return tableToMarkdown('Farsight DNSDB Lookup', out, headers=headers)
+    return tableToMarkdown(title, out, headers=headers)
```
Comment (orlichter1) [inline Packs/DNSDB/Integrations/DNSDB_v2/DNSDB_v2.py:540]: What happens if a search (either flex or rdata fetch) returns empty? Should be mentioned specifically that search returned empty.

File: Packs/DNSDB/Integrations/DNSDB_v2/DNSDB_v2.py [hunk 4]
```diff
@@ -239,13 +301,48 @@ def _query(self, path: str, limit: int = None, time_first_before: timeval = None
             )
         )
 
-        res = self._http_request('GET', path, params=params, stream=True, resp_type='response')
+        res = self._http_request('GET', path,
+                                 params=params,
+                                 stream=True, resp_type='response')
 
         if res.status_code == 404:
+            raise DemistoException('http 404')
```
Comment (orlichter1) [inline Packs/DNSDB/Integrations/DNSDB_v2/DNSDB_v2.py:309]: As you removed the 404 from the OK codes of the client, I believe this is not needed anymore.

**Review by orlichter1 — CHANGES_REQUESTED** [review]

File: Packs/DNSDB/pack_metadata.json [hunk 5]
```diff
@@ -2,7 +2,7 @@
     "name": "Farsight DNSDB",
     "description": "Query Farsight DNSDB service",
     "support": "partner",
-    "currentVersion": "2.0.1",
+    "currentVersion": "2.0.2",
```
[the remark this answers is not in the document]
Comment (hstern) [reply Packs/DNSDB/pack_metadata.json:5]: 61c0389771

File: Packs/DNSDB/ReleaseNotes/2_0_2.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Farsight DNSDB v2
+- Update to DNSDB API version 2 with Flexible Search
+- Upgraded the Docker image to demisto/python3:3.8.6.12176.
```
[the remark this answers is not in the document]
Comment (hstern) [reply Packs/DNSDB/ReleaseNotes/2_0_2.md:5]: 61c0389771

File: Packs/DNSDB/TestPlaybooks/playbook-DNSDB-Test.yml [hunk 7]
```diff
@@ -64,33 +64,34 @@ tasks:
...
     type: regular
     task:
-      id: e3f71089-12fa-43a5-8ba7-4d38b2795533
+      id: 5b696fe7-089b-4a5b-8bc1-92fe6f79da58
       version: -1
-      name: dnsdb-rdata
-      script: '|||dnsdb-rdata'
+      name: dnsdb-flex
+      description: DNSDB flex search
+      script: DNSDB_v2|||dnsdb-flex
```
[the remark this answers is not in the document]
Comment (hstern) [reply Packs/DNSDB/TestPlaybooks/playbook-DNSDB-Test.yml:74]: c17b34f346

File: Packs/DNSDB/Integrations/DNSDB_v2/README.md [hunk 8]
```diff
@@ -16,22 +16,123 @@ lookup rich, historical DNS information – either as playbook tasks or through
 API calls in the War Room – to access rdata and rrset records.  It was
 integrated and tested with version 1 of the DNSDB API.
```
[the remark this answers is not in the document]
Comment (hstern) [reply Packs/DNSDB/Integrations/DNSDB_v2/README.md:17]: c8b388b387

File: Packs/DNSDB/Integrations/DNSDB_v2/DNSDB_v2.py (hunk 4 above)
Comment (hstern) [reply Packs/DNSDB/Integrations/DNSDB_v2/DNSDB_v2.py:309]: 54cd28430b

File: Packs/DNSDB/Integrations/DNSDB_v2/DNSDB_v2.py (hunk 3 above)
Comment (hstern) [reply Packs/DNSDB/Integrations/DNSDB_v2/DNSDB_v2.py:540]: tableToMarkdown handles this nicely

image

File: Packs/DNSDB/Integrations/DNSDB_v2/DNSDB_v2_test.py [hunk 9]
```diff
@@ -1144,3 +1306,9 @@ def test_idna_dkim(self):
 
     def test_idna_email(self):
         assert DNSDB.parse_rdata("test@xn--frsight-exa.com.") == "test@fårsight.com."
+
+
+def _saf_wrap(records):
+    return '\n'.join(
+        ['{"cond":"begin"}'] + [f'{{"obj":{r}}}' for r in records] + ['{"cond":"succeeded"}']
+    )
```
[the remark this answers is not in the document]
Comment (hstern) [reply Packs/DNSDB/Integrations/DNSDB_v2/DNSDB_v2_test.py:1314]: 6cd9505ab9

**Review by orlichter1 — APPROVED** [review]

**Review by orlichter1 — APPROVED** [review]

**Pull request merged**